core/domain/collection_services.py

```diff
@@ -312,6 +313,51 @@ def get_multiple_collections_by_id(collection_ids, strict=True):
     return result
 
 
+def get_collection_and_collection_rights_by_id(collection_id, version=None):
+    """Returns a tuple for collection domain object and collection rights
+        object.
+
+    Args:
+        collection_id: str. Id of the collection.
+        version: str. Version of the collection.
+
+    Returns:
+        Tuple(collection object or None, collection rights object or None).
+    """
+    collection_memcache_key = _get_collection_memcache_key(
```

Comment:
No need to use memcache here. It gets complicated if the collection gets changed. Just datastore_services.fetch_multiple_...(...) should do.

Reply:
ok.